src/main/java/org/skriptlang/reflect/syntax/condition/elements/StructCustomCondition.java

```diff
@@ -0,0 +1,186 @@
+    info.ifPresent(dataTracker::setInfo);
+
+    dataTracker.addManaged(conditionHandlers);
+    dataTracker.addManaged(parserHandlers);
+    dataTracker.addManaged(usableSuppliers);
+    dataTracker.addManaged(parseSectionLoaded);
+  }
+
+  private SectionNode parseNode, checkNode;
+  private Runnable register;
+
+  @Override
+  public DataTracker<ConditionSyntaxInfo> getDataTracker() {
+    return dataTracker;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public boolean init(Literal<?>[] args, int matchedPattern, SkriptParser.ParseResult parseResult,
+                      EntryContainer entryContainer) {
+    customConditionsUsed = true;
+
+    SectionNode patterns = entryContainer.getOptional("patterns", SectionNode.class, false);
+    Script script = (parseResult.mark & 1) == 1 ? SkriptUtil.getCurrentScript() : null;
+
+    switch (matchedPattern) {
+      case 0:
+        String pattern = parseResult.regexes.get(0).group();
+        register = () -> register(ConditionSyntaxInfo.create(script, pattern, 1, false, false));
+        break;
+      case 1:
+        if (patterns == null) {
+          Skript.error("Custom conditions without inline patterns must have a patterns section.");
+          return false;
+        }
+
+        register = () -> {
+          int i = 1;
+          for (Node subNode : patterns) {
+            register(ConditionSyntaxInfo.create(script, subNode.getKey(), i++, false, false));
+          }
+        };
+        break;
+      case 2:
+        String property = parseResult.regexes.get(0).group();
+        String type = Arrays.stream(((Literal<ClassInfo<?>>) args[0]).getArray())
+            .map(ClassInfo::getCodeName)
+            .map(codeName -> {
+              boolean isPlural = Utils.getEnglishPlural(codeName).getSecond();
+
+              if (!isPlural) {
+                return Utils.toEnglishPlural(codeName);
+              }
+
+              return codeName;
+            })
+            .collect(Collectors.joining("/"));
+        register = () -> {
+          register(ConditionSyntaxInfo.create(script, "%" + type + "% (is|are) " + property, 1, false, true));
+          register(
+              ConditionSyntaxInfo.create(script, "%" + type + "% (isn't|is not|aren't|are not) " + property, 1, true, true));
+        };
+        break;
+    }
+
+    if (matchedPattern != 1 && patterns != null) {
+      Skript.error("Custom conditions with inline patterns may not have a patterns section.");
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean preLoad() {
+    super.preLoad();
+    register.run();
+    EntryContainer entryContainer = getEntryContainer();
+    parseNode = entryContainer.getOptional("parse", SectionNode.class, false);
+    SectionNode safeParseNode = entryContainer.getOptional("safe parse", SectionNode.class, false);
+    if (parseNode != null) {
+      if (safeParseNode != null) {
+        Skript.error("You can't have two parse sections");
```

Comment:
```suggestion
        Skript.error("A custom syntax element cannot contain both 'parse' and 'safe parse' entries");
```
It seems that this part of the code is repeated in each Structure. Is there any way we can abstract this out?